code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/NotificationBuilderTests.kt

```diff
@@ -0,0 +1,300 @@
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.ZeroBezelNotificationBuilder
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MockAEPPushTemplateDataProvider
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MockCarousalTemplateDataProvider
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MockProductCatalogTemplateDataProvider
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.MockTimerTemplateDataProvider
+import com.adobe.marketing.mobile.services.AppContextService
+import com.adobe.marketing.mobile.services.ServiceProvider
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkClass
+import io.mockk.mockkObject
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import io.mockk.verify
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class NotificationBuilderTests {
+
+    @Mock
+    private lateinit var application: Application
+    private lateinit var context: Context
+    private lateinit var alarmManager: AlarmManager
+    private lateinit var notificationManager: NotificationManager
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var broadcastReceiverClass: Class<out BroadcastReceiver>
+    private lateinit var notificationManagerCompat: NotificationManagerCompat
+
+    @Before
+    fun setUp() {
+        setupNotificationBuilderMocks()
+        setupApplicationMocks()
+        trackerActivityClass = mockkClass(Activity::class, relaxed = true).javaClass
+        broadcastReceiverClass = mockkClass(BroadcastReceiver::class, relaxed = true).javaClass
+        mockkObject(PendingIntentUtils)
+        mockkObject(PushTemplateImageUtils)
+    }
+
+    @After
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    private fun setupNotificationBuilderMocks() {
+        mockkObject(BasicNotificationBuilder)
+        mockkObject(ManualCarouselNotificationBuilder)
+        mockkObject(AutoCarouselNotificationBuilder)
+        mockkObject(InputBoxNotificationBuilder)
+        mockkObject(ZeroBezelNotificationBuilder)
+        mockkObject(ProductRatingNotificationBuilder)
+        mockkObject(ProductCatalogNotificationBuilder)
+        mockkObject(MultiIconNotificationBuilder)
+        mockkObject(TimerNotificationBuilder)
+        mockkObject(LegacyNotificationBuilder)
+    }
+
+    private fun setupApplicationMocks() {
+        mockkStatic(NotificationManagerCompat::class)
+        application = mockk<Application>(relaxed = true)
+        context = mockk<Context>(relaxed = true)
+        alarmManager = mockk<AlarmManager>(relaxed = true)
```

Comment:
Do we still need L104 through L111 since there's no remind later functionality here?

Reply:
timer notification builder checks for exact alarm permission so it's still needed.